### ISSUE TITLE
Empty EOL comment causes exception

### ISSUE THREAD

**Issue by jaydunning** [jsdoc3/jsdoc#920]
An end-of-line comment immediately followed by a newline results in a StringIndexOutOfBoundsException. 

Tested in current, 3.3.0-beta1, v3.3.0-alpha13, on Windows 7, Java 1.6.0_38.

Although I found it in the middle of a file, the test case can be as simple as a one-line .js file.

````
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: -2
        at java.lang.String.substring(Unknown Source)
        at org.jsdoc.AstBuilder.processComment(AstBuilder.java:600)
        at org.jsdoc.AstBuilder.processNode(AstBuilder.java:391)
        at org.jsdoc.AstBuilder.access$000(AstBuilder.java:31)
        at org.jsdoc.AstBuilder$1.visit(AstBuilder.java:308)
        ...
````

**Comment by hegemonic** [comment]
Thanks for the heads up. I'll take a look at this when I have access to my Windows box.

**Comment by hegemonic** [comment]
I wasn't able to reproduce this issue, but I've made a change that should prevent it from happening. Please try the latest version from GitHub master and see if it works for you. I'll reopen this issue if you're still seeing the bug.

**Comment by jaydunning** [comment]
Bug's still there. The cause of the problem is that the second argument to String.substring is the end index, not the length. That expression could actually be
````
    comment.substring(2);
````

 I had to decompile AstBuilder to figure that out. Where's the source?

**Comment by hegemonic** [comment]
The source lives in https://github.com/jsdoc3/rhino.

Are you still getting exactly the same exception that you were getting before? If not, can you please provide an updated stack trace?

Either way, can you please provide a source file that I can use to reproduce this issue?

**Comment by jaydunning** [comment]
Thanks. I knew I'd seen it somewhere.

Source file for test case (1 line; EOL comment immediately followed by newline followed by EOF):
````
//
````

Same exception.

The exception occurs at AstBuilder.java, line 600:
````
		info.put("value", comment.substring(2, Math.max(0, comment.length() - 2)));
````
It should be something like:
````
        String strippedComment = (
             "//".equals(comment.substring(0, 2))
             ? comment.length() > 2 ? comment.substring(2) : ""
             : comment.substring(2, comment.length() - 2)
        );
        info.put("value", strippedComment);
````

**Comment by hegemonic** [comment]
Thanks for clarifying. The issue is now fixed on master and releases/3.3.
